Yoast/wordpress-seo
# [UI library] Add toast element

## Context


* We want to add a generic `Toast` element to the UI Library in order to render customizable notifications.

## Summary


This PR can be summarized in the following changelog entry:

* [@yoast/ui-library 0.0.1] Adds a `Toast` element to the Yoast UI library.

## Relevant technical choices:

* The `Notification` component has been refactored to use the new `Toast` element.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

<details>
  <summary>Test the Settings notification</summary>

* Go to `Yoast SEO` -> `Settings`
  * change any setting and hit `Save changes`
  * verify the following notification appears
  <img width="356" alt="Screenshot 2024-05-24 at 12 30 01" src="https://github.com/Yoast/wordpress-seo/assets/68744851/b45c9eaf-a8cf-4cc5-b1ef-d1d095b881fe">
  
  * wait until it closes automatically
  * change another setting, and when the notification appears click on the close button
    * verify the notification closes immediately 
* Go to `Yoast SEO` -> `Settings` -> `Site connections`
  * in the `Google` input field type `&&&&` and click on `Save changes`
  * verify:
    * the following notification appears
    <img width="414" alt="Screenshot 2024-05-24 at 12 32 23" src="https://github.com/Yoast/wordpress-seo/assets/68744851/69f818fb-0631-4584-8eff-3d5ff76f176c">

    * the notification stays open until you close it 

</details>

<details>
  <summary>Test the AI Generator notification</summary>

* With a MySQL client open the `wp_usermeta` table and search for the tuple where `user_id` = your user id and `meta_key` = `_yoast_alerts_dismissed`
* if the following string is present in the `meta_value` field:`s:26:"wpseo_premium_ai_generator";b:1;`, remove it and update the array length accordingly (or remove the whole `meta_value` content if you don't care about the notifications you previously dismissed)
* Activate Yoast SEO Premium
* Edit a post in the block editor
  * make sure
    * it has little to no content
    *  it has a keyphrase
  * Click on `Use AI` and verify the following notification appears
   
  <img width="348" alt="Screenshot 2024-05-24 at 12 53 53" src="https://github.com/Yoast/wordpress-seo/assets/68744851/50c02d57-f4d0-4c26-bfce-a7963a6c6f9b">
  
  * verify that either clicking on the close or `Dismiss` button closes the notification
  * click on `Use AI` again
    * verify the notification is shown again

</details>

<details>
  <summary>Test the UI-Library storybook</summary>

* In the terminal got to the wordpress-seo/packages/ui-library directory
  * type `yarn storybook`: after some time the StoryBook page should appear in your browser
  * go to the `Toaster` sub-section in the `Elements` section and verify
    * all the stories ( the `Toaster` factory, `with title`, `with description` and `with close button` ) render properly
    * you can correctly interact with the toaster
 
</details>

#### Relevant test scenarios


### Test instructions for QA when the code is in the RC



## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

* The above testing instructions cover the possible regressions.

## UI changes


## Other environments


## Documentation


## Quality assurance


## Innovation


Fixes [#21396](https://github.com/Yoast/wordpress-seo/issues/21396)
